file.go

```diff
@@ -0,0 +1,198 @@
+package nightfall
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// An object containing configuration that describes how to scan a file. Since the file is scanned asynchronously,
+// the results from the scan are delivered to the provided webhook URL. The scan configuration may contain both
+// inline detection rule definitions and UUID's referring to existing detection rules (up to 10 of each).
+type ScanPolicy struct {
+	WebhookURL         string          `json:"webhookURL"`
+	DetectionRules     []DetectionRule `json:"detectionRules"`
+	DetectionRuleUUIDs []uuid.UUID     `json:"detectionRuleUUIDs"`
+}
+
+// A container for a request to scan a file that was uploaded via the Nightfall API. Exactly one of
+// PolicyUUID or Policy should be provided.
+type ScanFileRequest struct {
+	PolicyUUID       *uuid.UUID    `json:"policyUUID"`
+	Policy           *ScanPolicy   `json:"policy"`
+	RequestMetadata  string        `json:"requestMetadata"`
+	Content          io.Reader     `json:"-"`
+	ContentSizeBytes int64         `json:"-"`
+	Timeout          time.Duration `json:"-"`
+}
+
+// The object returned by the Nightfall API when an (asynchronous) file scan request was successfully triggered.
+type ScanFileResponse struct {
+	ID      uuid.UUID `json:"id"`
+	Message string    `json:"message"`
+}
+
+type fileUploadResponse struct {
+	ID            uuid.UUID `json:"id"`
+	FileSizeBytes int64     `json:"fileSizeBytes"`
+	ChunkSize     int64     `json:"chunkSize"`
+	MIMEType      string    `json:"mimeType"`
+}
+
+type fileUploadRequest struct {
+	FileSizeBytes int64 `json:"fileSizeBytes"`
+}
+
+// A convenience method that abstracts the details of the multi-step file upload and scan process. In other words,
+// calling this method for a given file is equivalent to (1) manually initializing a file upload session,
+// (2) uploading all chunks of the file, (3) completing the upload, and (4) triggering a scan of the file.
+//
+// The maximum allowed ContentSizeBytes is dependent on the terms of your current
+// Nightfall usage plan agreement; check the Nightfall dashboard for more details.
+//
+// This method consumes the provided reader, but it does not close it; closing remains
```

Comment:
`io.Reader` does not need to be closed in right? unless there's some advantage to us changing the interface to `io.ReadCloser`, not sure

Reply:
it does need to be closed, but i feel like closing is the caller's responsibility, similar to how it is with filesage